test/e2e/pivoting_test.go

```diff
@@ -113,32 +111,29 @@ func pivoting() {
 	})
 	Expect(controlPlane).ToNot(BeNil())
 
-	By("Check if BMH is in provisioned state")
-	Eventually(func(g Gomega) {
-		bmhList := &bmov1alpha1.BareMetalHostList{}
-		g.Expect(targetCluster.GetClient().List(ctx, bmhList, client.InNamespace(namespace))).To(Succeed())
-		for _, bmh := range bmhList.Items {
-			g.Expect(bmh.WasProvisioned()).To(BeTrue())
-		}
-	}, e2eConfig.GetIntervals(specName, "wait-object-provisioned")...).Should(Succeed())
+	By("Check that BMHs are in provisioned state")
+	waitForNumBmhInState(ctx, bmov1alpha1.StateProvisioned, waitForNumInput{
+		Client:    targetCluster.GetClient(),
+		Options:   []client.ListOption{client.InNamespace(namespace)},
+		Replicas:  4,
+		Intervals: e2eConfig.GetIntervals(specName, "wait-object-provisioned"),
+	})
 
 	By("Check if metal3machines become ready.")
-	Eventually(func(g Gomega) {
-		m3Machines := &infrav1.Metal3MachineList{}
-		g.Expect(targetCluster.GetClient().List(ctx, m3Machines, client.InNamespace(namespace)))
-		for _, m3Machine := range m3Machines.Items {
-			g.Expect(m3Machine.Status.Ready).To(BeTrue())
-		}
-	}, e2eConfig.GetIntervals(specName, "wait-object-provisioned")...).Should(Succeed())
-
-	By("Check if machines become running.")
-	Eventually(func(g Gomega) {
-		machines := &clusterv1.MachineList{}
-		g.Expect(targetCluster.GetClient().List(ctx, machines, client.InNamespace(namespace))).To(Succeed())
-		for _, machine := range machines.Items {
-			g.Expect(strings.EqualFold(machine.Status.Phase, "running")).To(BeTrue())
-		}
-	}, e2eConfig.GetIntervals(specName, "wait-machine-running")...).Should(Succeed())
+	waitForNumMetal3MachinesReady(ctx, waitForNumInput{
+		Client:    targetCluster.GetClient(),
+		Options:   []client.ListOption{client.InNamespace(namespace)},
+		Replicas:  4,
+		Intervals: e2eConfig.GetIntervals(specName, "wait-object-provisioned"),
```

Comment:
Sure, any suggestion?

Reply:
I just wanted independent interval name for m3m ready than the one used for bmh provisioned as machine running has its own interval but since the time needed for `bmh to provisioned ~= machine running ~= m3m ready` maybe just rename the intervals` [wait-machine-running, wait-object-provisioned] `with `wait_bmh_provisioned_machine_running_m3m_ready`and use it for all of them I noted also we usually wait for machine running or m3m ready after ensuring bmh provisioned so the time needed here can be reduced if we want the test to fail without waiting time waiting for a long time but on the other hand this will not optimize time needed for for the test to pass

Reply:
I'm not sure about this. It would be very long to have `wait_bmh_provisioned_machine_running_m3m_ready` and if we anyway use it for many different things then I think it is fine to let it be `wait-object-provisioned`. It is currently set to 10m with 10s interval, which should be fine for machine/metal3machines after the BMH is provisioned.

Reply:
that's fine with me also